### PR TITLE
feat(editor): added shopify elements and placeholders for them

### PR DESCRIPTION
**Pull request by nicolaisirghi**
### Changelog
- New: Added shopify elements in editor

### CODE REVIEW

File: packages/core/src/core.ts [hunk 1]
```diff
@@ -497,6 +528,36 @@ export const Core: Init<HtmlOutputType> = (token, config, cb) => {
...
+
+            if (typeof handler === "function") {
+              const res = (r: any) => {
+                iframeWindow.postMessage(actions.getCollectionItemsIdsRes(r, uid), targetOrigin);
+              };
+              const rej = (r: string) => {
+                iframeWindow.postMessage(actions.getCollectionItemsIdsRej(r, uid), targetOrigin);
+              };
+
+              handler?.(res, rej, extra);
```
Comment (maxval1) [inline packages/core/src/core.ts:558]: remove the `?`

**Review by maxval1 — CHANGES_REQUESTED** [review]

File: packages/core/src/types/types.ts [hunk 2]
```diff
@@ -48,10 +60,15 @@ export interface Extension {
 
 export type HtmlOutputType = "html" | "json";
 
+interface PageDataType extends Record<string, unknown> {
+  id: Literal;
+  reference?: Reference;
```
[the remark this answers is not in the document]
Comment (nicolaisirghi) [reply packages/core/src/types/types.ts:65]: Tell me when will be needed to discuss

File: packages/demo-nextjs/src/lib/editorConfig/index.ts [hunk 3]
```diff
@@ -9,7 +9,8 @@ export const getProjectData = (model: Project): Config["projectData"] => {
 
 export const getPageData = (model: Item): Config["pageData"] => {
   const id = model._id;
+  const reference = JSON.parse(model.config?.reference ?? "{}");
```
Comment (maxval1) [inline packages/demo-nextjs/src/lib/editorConfig/index.ts:12]: rollback

File: packages/thirdparty-widgets/src/Cart/export/addToCart.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import { getAddToCartData, reinitializeCarts, showUserErrors } from "./utils";
```
Comment (maxval1) [inline packages/thirdparty-widgets/src/Cart/export/addToCart.ts:1]: need to review

File: packages/thirdparty-widgets/src/Cart/export/constants.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export const DEFAULT_TITLE = "Default Title";
```
Comment (maxval1) [inline packages/thirdparty-widgets/src/Cart/export/constants.ts:1]: review

File: packages/thirdparty-widgets/src/Cart/export/index.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import { processAddToCart } from "./addToCart";
+import { closeCart, openCart, reinitializeCart } from "./utils";
+import { getCart } from "./requests";
```
Comment (maxval1) [inline packages/thirdparty-widgets/src/Cart/export/index.ts:3]: review

File: packages/thirdparty-widgets/src/Cart/export/requests.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import { Variant } from "demo-nextjs/src/lib/db/shopify/types";
+import axios from "axios";
+
+export const addItemToCart = ({
```
Comment (maxval1) [inline packages/thirdparty-widgets/src/Cart/export/requests.ts:4]: review

File: packages/thirdparty-widgets/src/Cart/export/utils.ts [hunk 8]
```diff
@@ -0,0 +1,293 @@
+import { CartItem } from "../types";
+import { deleteCartItem, updateCartItem } from "./requests";
+import { Amount, InitialiseCartData } from "./types";
+import { DEFAULT_TITLE } from "./constants";
+
+// Utility functions
+const toggleClass = (element: HTMLElement, className: string, add: boolean) =>
```
Comment (maxval1) [inline packages/thirdparty-widgets/src/Cart/export/utils.ts:7]: review

File: packages/thirdparty-widgets/src/components/ThemeIcon/index.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+}: {
+  className?: string;
+  href: string;
+}): JSX.Element => (
+  <svg className={className}>
+    <use href={href} />
+  </svg>
+);
+
+const ThemeIconPreview = (props: Props): JSX.Element => {
```
Comment (maxval1) [inline packages/thirdparty-widgets/src/components/ThemeIcon/index.tsx:18]: need to review
1. Documentation
2. How we can use it from the Editor(edit, view mode)

**Review by maxval1 — CHANGES_REQUESTED** [review]
Small PR

File: packages/demo-nextjs/src/placeholders/types/BuilderPlaceholder.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+    const decodedContent = atob(content);
+    const match = decodedContent.match(/\{\{(.*?)\}\}|(\w+)/);
+    const placeholderName = match?.[1] || match?.[2];
+
+    const attrStr = Object.entries(attrs)
+      .map(([key, value]) => `${key}='${value}'`)
+      .join(" ");
+
+    return `{{${placeholderName} ${attrStr}}}`;
```
Comment (maxval1) [inline packages/demo-nextjs/src/placeholders/types/BuilderPlaceholder.ts:28]: extract plachodelr via sdk

**Review by maxval1 — CHANGES_REQUESTED** [review]

**Pull request merged**